src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -484,4 +489,76 @@ else if (tokenType == TokenTypes.LITERAL_PRIVATE) {
         return illegalClassNames;
     }
 
+    /**
+     * Strip initial newline and preceding whitespace on each line from text block content.
+     * In order to be consistent with how javac handles this task, we have modeled this
+     * implementation after the code from:
+     * github.com/openjdk/jdk14u/blob/master/src/java.base/share/classes/java/lang/String.java
+     *
+     * @param textBlockContent the actual content of the text block.
+     * @return string consistent with javac representation.
+     */
+    public static String stripIndentAndInitialNewLineFromTextBlock(String textBlockContent) {
+        final String contentWithInitialNewLineRemoved =
+            ALL_NEW_LINES.matcher(textBlockContent).replaceFirst("");
+        final List<String> lines =
+            Arrays.asList(ALL_NEW_LINES.split(contentWithInitialNewLineRemoved));
+        final int indent = getSmallestIndent(lines);
+        final String suffix = "";
+
+        return lines.stream()
+                .map(line -> stripIndentFromLine(line, indent))
+                .collect(Collectors.joining(System.lineSeparator(), "", suffix));
+    }
+
+    /**
+     * Helper method for stripIndentAndInitialNewLineFromTextBlock, strips correct indent
+     * from string, or returns empty string if no text.
+     *
+     * @param line the string to strip indent from
+     * @param indent the amount of indent to remove
+     * @return modified string with removed indent, or empty string.
+     */
+    private static String stripIndentFromLine(String line, int indent) {
+        final int firstNonWhitespace = indexOfNonWhitespace(line);
+        final int lastNonWhitespace = line.length();
+        final int incidentalWhitespace = Math.min(indent, firstNonWhitespace);
+
+        return line.substring(incidentalWhitespace, lastNonWhitespace);
+    }
+
+    /**
+     * Helper method for stripIndentAndInitialNewLineFromTextBlock, to determine the smallest
+     * indent in a text block string literal.
+     *
+     * @param lines list of actual text block content, split by line.
+     * @return number of spaces representing the smallest indent in this text block.
+     */
+    private static int getSmallestIndent(List<String> lines) {
+        int indent = Integer.MAX_VALUE;
+        for (String line : lines) {
+            final int leadingWhitespace = indexOfNonWhitespace(line);
+            indent = Integer.min(indent, leadingWhitespace);
+        }
+        return indent;
+    }
+
+    /**
+     * Helper method to find the index of the first non-whitespace character in a string.
+     *
+     * @param line the string to find the first index of a non-whitespace character for.
+     * @return the index of the first non-whitespace character.
+     */
+    private static int indexOfNonWhitespace(String line) {
```

Comment:
This method is similar to `CommonUtil::isBlank`. May be it should be rewritten as
```java
    public static boolean isBlank(String value) {
        return indexOfNonWhitespace(value) < 0;
    }
```
?

Reply:
Can you explain a bit more? I am using `indexOfNonWhitespace()` to return an int value to compare with `indent` to get the substring from `line`.

Reply:
Just take a look at [isBlank](https://github.com/checkstyle/checkstyle/blob/2e8c277c181487d81231170ff4b959edc008114e/src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java#L651) method. It is almost same as the new method `indexOfNonWhitespace`.  I want to get rid of duplication.

Reply:
Ok, I misunderstood what you were asking, I think.  I have refactored `isBlank()`, and moved `indexOfNonWhitespace()` to CommonUtil.java.